bluesky/interactive.py

```diff
@@ -0,0 +1,130 @@
+import matplotlib.widgets as mwidgets
+from bluesky.plans import outer_product_scan
+from collections import ChainMap
+
+
+class ROIPlanCreator:
+    def __init__(self, ax):
+        '''Base class for interactively creating plans in a 2D bounding box.
+
+        Parameters
+        ----------
+        ax : matplotlib.axes.Axes
+            The axes to install the widget on
+        '''
+        self.ax = ax
+        self.widget = mwidgets.RectangleSelector(
+            self.ax, self._onselect, useblit=True, interactive=True)
+        self._pt1 = self._pt2 = None
+
+    def _onselect(self, pt1, pt2):
+        self._pt1 = pt1
+        self._pt2 = pt2
+
+    @property
+    def last_rect(self):
+        '''data coordinates of the first and second corners.
+
+        Coordinates are lower left and upper right.
+
+        Returns
+        -------
+        points : tuple
+           ((x1, y1), (x2, y2))
+        '''
+        # TODO deal with ordering issues
```

Comment:
that comment should not have survived.  I was not sure if mpl return the points in the order you drew them or (bottom left, top right) and it turns out it always is (BL, UR).